python-stubs/dolphin_debug.pyi

```diff
@@ -0,0 +1,274 @@
+        - "BreakOnWrite": bool - whether the bp should trigger on memory write
+        - "LogOnHit": bool - whether Dolphin should log when the bp is hit
+        - "BreakOnHit": bool - whether execution should pause when the bp is hit
+        - "Condition": str - set a condition for the breakpoint to be hit.
+                             GPRs can be referenced by r0-r31 and FPRs by f0-f31.
+                             You can also use lr, ctr, and pc.
+    
+    """
+
+
+def remove_memory_breakpoint(addr: int, /) -> None:
+    """
+    Remove the memory breakpoint at the specified addr.
+    
+    :param addr: address of the breakpoint to remove
+    """
+
+
+
+
+
+
+
+
+
+
+
+
+def read_u8(addr: int, /) -> int:
+    """
+    Reads 1 byte as an unsigned integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_u16(addr: int, /) -> int:
+    """
+    Reads 2 bytes as an unsigned integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_u32(addr: int, /) -> int:
+    """
+    Reads 4 bytes as an unsigned integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_u64(addr: int, /) -> int:
+    """
+    Reads 8 bytes as an unsigned integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_s8(addr: int, /) -> int:
+    """
+    Reads 1 byte as a signed integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_s16(addr: int, /) -> int:
+    """
+    Reads 2 bytes as a signed integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_s32(addr: int, /) -> int:
+    """
+    Reads 4 bytes as a signed integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_s64(addr: int, /) -> int:
+    """
+    Reads 8 bytes as a signed integer.
+
+    :param addr: memory address to read from
+    :return: value as integer
+    """
+
+
+def read_f32(addr: int, /) -> float:
+    """
+    Reads 4 bytes as a floating point number.
+
+    :param addr: memory address to read from
+    :return: value as floating point number
+    """
+
+
+def read_f64(addr: int, /) -> float:
+    """
+    Reads 8 bytes as a floating point number.
+
+    :param addr: memory address to read from
+    :return: value as floating point number
+    """
+
+
+def read_bytes(addr: int, size: int, /) -> bytearray:
+    """
+    Reads size bytes and outputs a bytearray of length size.
+    
+    :param addr: memory address to start reading from
+    :param size: number of bytes to read
+    :return: bytearray containing the read bytes
+    """
+
+
+def invalidate_icache(addr: int, size: int, /) -> None:
+    """
+    Invalidates JIT cached code between the address and address + size, \
```

Comment:
Is the `\` needed here?

Reply:
That's the auto formatter trying to do its job.

Reply:
No that's me fat-fingering the backslash key next to the enter key